relay-server/src/metrics_extraction/event.rs

```diff
@@ -497,16 +500,21 @@ mod tests {
         let mut event = Annotated::from_json(json).unwrap();
         let features = FeatureSet(BTreeSet::from([Feature::SpanMetricsExtraction]));
 
-        // Normalize first, to make sure that all things are correct as in the real pipeline:
+        // Validate and normalize first, to make sure that all things are correct as in the real pipeline:
+        let res = validate_transaction(&event, &TransactionValidationConfig::default());
+        assert!(res.is_ok());
+
+        let res = validate_event_timestamps(&mut event, &EventValidationConfig::default());
+        assert!(res.is_ok());
+
```

Comment:
I would try to run the test without this, and remove validation if it passes. Ideally tests only run the code that they're actually trying to test. See [internal guidelines](https://www.notion.so/sentry/Relay-Code-and-Architecture-4fac481cf27741aba54d39594f1cae58?pvs=4#d8c37053239e46fbb319a06428fac18d).

```suggestion
        // Validate and normalize first, to make sure that all things are correct as in the real pipeline:
        let res = validate_transaction(&event, &TransactionValidationConfig::default());
        assert!(res.is_ok());

        let res = validate_event_timestamps(&mut event, &EventValidationConfig::default());
        assert!(res.is_ok());

```